dd-java-agent/instrumentation/jedis-1.4/src/main/java/datadog/trace/instrumentation/jedis/JedisInstrumentation.java

```diff
@@ -34,9 +34,8 @@ public JedisInstrumentation() {
 
   @Override
   public ElementMatcher<ClassLoader> classLoaderMatcher() {
-    return classLoaderHasNoResources("redis/clients/jedis/commands/ProtocolCommand.class")
-        // Optimization for expensive typeMatcher.
-        .and(not(classLoaderHasNoResources("redis/clients/jedis/Protocol.class")));
+    // Avoid matching 3.x
+    return not(hasClassesNamed("redis.clients.jedis.commands.ProtocolCommand"));
```

Comment:
```suggestion
    // Avoid matching 3.x
    return not(hasClassesNamed("redis.clients.jedis.commands.ProtocolCommand"))
        // Optimization for expensive typeMatcher.
        .and(hasClassesNamed("rediis.clients.jedis.Protocol"));
```

The non-double-negative part is missing

Reply:
This is actually intentional.  `JedisInstrumentation` was doing a simple `named` matcher, so adding the classload matcher in unnecessary.  (It was added by accident.)